src/Domain/FunctionsOfTime/Tags.cpp

```diff
@@ -3,31 +3,92 @@
 
 #include "Domain/FunctionsOfTime/Tags.hpp"
 
+#include <cstddef>
 #include <memory>
+#include <optional>
+#include <tuple>
 
 #include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
 #include "Domain/Domain.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/ReadSpecThirdOrderPiecewisePolynomial.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 
-namespace domain {
-namespace Tags {
+namespace domain::Tags {
 template <size_t Dim>
-auto InitialFunctionsOfTime<Dim>::create_from_options(
+auto InitialFunctionsOfTime<Dim, true>::create_from_options(
+    const std::unique_ptr<::DomainCreator<Dim>>& domain_creator,
+    const std::optional<std::string>& function_of_time_file,
+    const std::optional<std::map<std::string, std::string>>&
+        function_of_time_name_map) noexcept -> type {
+  if (function_of_time_file and function_of_time_name_map) {
+    // Currently, only support order 3 piecewise polynomials.
+    // This could be generalized later, but the SpEC functions of time
+    // that we will read in with this action will always be 3rd-order
+    // piecewise polynomials
+    constexpr size_t max_deriv{3};
+    std::unordered_map<std::string,
+                       domain::FunctionsOfTime::PiecewisePolynomial<max_deriv>>
+        spec_functions_of_time{};
+    importers::read_spec_third_order_piecewise_polynomial(
+        make_not_null(&spec_functions_of_time), *function_of_time_file,
+        *function_of_time_name_map);
+
+    std::unordered_map<std::string,
+                       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+        functions_of_time{domain_creator->functions_of_time()};
+    for (auto spec_and_spectre_names : *function_of_time_name_map) {
```

Comment:
I didn't know about structured bindings...thanks!

Reply:
@nilsleiffischer @nilsdeppe I get a warning (error in the tests) about undefined variables, because I actually only use spectre_name, not spec_name. Is there a way with structured bindings to tell the compiler I'm not using the first one? If not, should I change it back to the way it was, instead of using structured bindings?

Reply:
just do a:
```cpp
(void)spec_name;
```
to "use" it. That's what I always do. I find the structured bindings sufficiently easier to read that the cast is worth it.